### PR TITLE
Overhaul navigation

### PR DESCRIPTION
**Pull request by dabreegster**
#27 

This centralizes a "sitemap", using it to set page titles, take over the breadcrumb work, and handle previous/next pages. Main TODOs:

- improve styling
- handle the branching logic in the route check. We can probably just toggle off some of the entries after the user decides street vs path check.

### CODE REVIEW

File: src/lib/nav.ts [hunk 1]
```diff
@@ -0,0 +1,315 @@
+// Produced manually with great care
```
Comment (dabreegster) [inline src/lib/nav.ts:1]: In some intermediate commits, I was writing a tree structure and flattening it, but the flat structure is nicer to use for all of the methods, and writing the tree isn't any easier than writing the flat list

File: src/routes/+layout.svelte [hunk 2]
```diff
@@ -3,14 +3,66 @@
...
   This is an experimental tool. Use only for internal testing.
 </WarningText>
 
 <div class="govuk-prose">
+  <Breadcrumbs
+    links={getBreadcrumbLinks(pagePath)}
+    current={getTitle(pagePath)}
+  />
+
+  <div style="display: flex; justify-content: space-between">
```
Comment (dabreegster) [inline src/routes/+layout.svelte:37]: Styling here TBD

File: src/routes/cross_section/results_summary/+page.svelte
Comment (dabreegster) [inline src/routes/cross_section/results_summary/+page.svelte:1]: Stubbing out empty pages so things work

File: src/routes/+layout.svelte (hunk 2 above)
Comment (Pete-Y-CS) [inline src/routes/+layout.svelte:37]: I guess making this a component is a bit of a faff but we might want the same thing at both top and bottom right?

**Review by Pete-Y-CS — APPROVED** [review]
Tiny comment that prev/next probably wants to be a component but generally this looks good!

File: src/lib/nav.ts (hunk 1 above)
Comment (dabreegster) [reply src/lib/nav.ts:1]: Sure, done

File: src/routes/+layout.svelte (hunk 2 above)
Comment (dabreegster) [reply src/routes/+layout.svelte:37]: I was going to, but then not sure if the style will be different between the two. Top has the breadcrumbs, maybe we want to put on the sides of that? Or maybe that would jump around on different pages even more

**Review by Pete-Y-CS — APPROVED** [review]

**Pull request merged**